presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,326 @@
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.metadata.FunctionExtractor.extractFunctions;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+
+public class TestGeoQueries
+        extends AbstractTestQueryFramework
+{
+    public TestGeoQueries()
+    {
+        super(TestGeoQueries::createLocalQueryRunner);
+    }
+
+    @Test
+    public void testSTPoint()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_point(1, 4))", "select 'POINT (1 4)'");
+    }
+
+    @Test
+    public void testSTLineString()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_line_string('linestring(1 1, 2 2, 1 3)'))", "select 'LINESTRING (1 1, 2 2, 1 3)'");
+    }
+
+    @Test
+    public void testSTPolygon()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_polygon('polygon ((1 1, 1 4, 4 4, 4 1))'))", "select 'POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))'");
+    }
+
+    @Test
+    public void testSTArea()
+            throws Exception
+    {
+        assertQuery("select st_area(st_geometry_from_text('polygon ((2 2, 2 6, 6 6, 6 2))'))", "select 16.0");
+    }
+
+    @Test
+    public void testSTCentroid()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_centroid(st_geometry_from_text('polygon ((1 1, 1 4, 4 4, 4 1))')))", "select 'POINT (2.5 2.5)'");
+    }
+
+    @Test
+    public void testSTCoordinateDimension()
+            throws Exception
+    {
+        assertQuery("select st_coordinate_dimension(st_geometry_from_text('polygon ((1 1, 1 4, 4 4, 4 1))'))", "select 2");
+    }
+
+    @Test
+    public void testSTDimension()
+            throws Exception
+    {
+        assertQuery("select st_dimension(st_geometry_from_text('polygon ((1 1, 1 4, 4 4, 4 1))'))", "select 2");
+    }
+
+    @Test
+    public void testSTIsClosed()
+            throws Exception
+    {
+        assertQuery("select st_is_closed(st_geometry_from_text('linestring(1 1, 2 2, 1 3, 1 1)'))", "select true");
+    }
+
+    @Test
+    public void testSTIsEmpty()
+            throws Exception
+    {
+        assertQuery("select st_is_empty(st_geometry_from_text('POINT (1.5 2.5)'))", "select false");
+    }
+
+    @Test
+    public void testSTLength()
+            throws Exception
+    {
+        assertQuery("select st_length(st_geometry_from_text('linestring(0 0, 2 2)'))", "select 2.8284271247461903");
+        assertQuery("select st_length(st_geometry_from_text('polygon ((1 1, 1 4, 4 4, 4 1))'))", "select 12.0");
+    }
+
+    @Test
+    public void testSTMax()
+            throws Exception
+    {
+        assertQuery("select st_max_x(st_geometry_from_text('linestring(8 4, 4 8)'))", "select 8.0");
+        assertQuery("select st_max_y(st_geometry_from_text('linestring(8 4, 4 8)'))", "select 8.0");
```

Comment:
nit: consider testing using a linestring with different max_x and max_y; otherwise a typo in the implementation which makes max_x and max_y always return same value will go unnoticed